udata/core/dataset/models.py

```diff
@@ -151,13 +151,22 @@ def check_availability(self, group):
     def is_available(self):
         return self.check_availability(group=None)
 
+    @property
+    def permalink(self):
+        '''
+        Permanent link to the latest version of this resource.
+
+        If this resource is updated and `url` change, this property won't.
+        '''
+        return url_for('datasets.resource', id=self.id, _external=True)
+
     @cached_property
     def json_ld(self):
 
         result = {
             '@type': 'DataDownload',
             '@id': str(self.id),
-            'url': self.url,
+            'url': self.permalink,
```

Comment:
BTW, this is exactly the use case, the reason why there is an extra `contentUrl`: `url` is e generic URL also used as a an object identifier (can be a presentation page of a music track) whereas `contentUrl` is a direct access to the content (in case of a music track, it will be the direct access to the binary file ie. mp3 or whatever)